### DISCUSSION TITLE
Anyone integrated Wordpress with Budibase ?

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#6399] [Help] [answered]
One of the things i need to implement, is for my Wordpress site to pull data from BB and also post Form submissions to BB API.
Has anyone done something like this ? 
Any tips / samples or ideas before i dive into coding it ?

**Comment by poirazis** [comment]
This may sound like a really stupid question but, how do i get the 'tableId" required by the API to insert a new row into a table ?

PS I got it from the query string from the admin site , but is there another way ?
It might be usefull to have a "Copy Table ID" similar to "Copy App ID" in the table dropdown menu.

**Comment by melohagan** [comment, marked as answer]
Hi @poirazis 

There's no way that I can think of posting form submissions i.e. including triggering validation and other form actions, from the API, but you can, as you mention, insert table rows:
https://docs.budibase.com/reference/post_tables-tableid-rows

You can pull table rows from a Budibase app with: https://docs.budibase.com/reference/post_tables-tableid-rows-search

You can get the table id from the URL (after the **/table/**) when in the data view of Budibase of a particular table. Budibase table ids should start with **ta_**
 
![Screenshot 2022-06-21 at 10 25 12](https://user-images.githubusercontent.com/101575380/174766109-5b6dbd85-8850-4799-9908-59dc696ad99e.png)
![Screenshot 2022-06-21 at 10 24 52](https://user-images.githubusercontent.com/101575380/174766032-a3beaf30-f615-4bd8-9cb1-9c3a9114b477.png)
 
Alternatively, you can search for tables: https://docs.budibase.com/reference/post_tables-search
It states that the name is required, but you actually can leave it blank to retrieve all internal tables for a given app. Then you can grab the **_id** for each returned element:
<img width="610" alt="Screenshot 2022-06-21 at 10 28 59" src="https://user-images.githubusercontent.com/101575380/174766854-c832b348-61ef-4402-b4c2-b760f4f680db.png">

**Reply by poirazis** [reply]
Thank you @melohagan !
I am doing Wordpress end Form validations before sending them over to BB so I am just inserting data that are later validated / processed with Automations.

I am also retrieving data for display purposes using a free plugin[ (wpgetapi] (https://wordpress.org/plugins/wpgetapi/) for anyone else looking for a way to get data from BB into Wordpress.
